src/pcp2opentelemetry/pcp2opentelemetry.py

```diff
@@ -0,0 +1,675 @@
+
+        results = self.pmconfig.get_ranked_results(valid_only=True)
+        def get_type_string(desc):
+            """ Get metric type as string """
+            if desc.contents.type == pmapi.c_api.PM_TYPE_32:
+                mtype = "32"
+            elif desc.contents.type == pmapi.c_api.PM_TYPE_U32:
+                mtype = "u32"
+            elif desc.contents.type == pmapi.c_api.PM_TYPE_64:
+                mtype = "64"
+            elif desc.contents.type == pmapi.c_api.PM_TYPE_U64:
+                mtype = "u64"
+            elif desc.contents.type == pmapi.c_api.PM_TYPE_FLOAT:
+                mtype = "float"
+            elif desc.contents.type == pmapi.c_api.PM_TYPE_DOUBLE:
+                mtype = "float"
+            elif desc.contents.type == pmapi.c_api.PM_TYPE_STRING:
+                mtype = "string"
+            else:
+                mtype = "unknown"
+            return mtype
+
+        def attribute_converter(a_dictionary):
+            entry_list = []
+            for key, value in a_dictionary.items():
+                new_dictionary = {"key": key, "value": {"stringValue": value}}
+                entry_list.append(new_dictionary)
+
+            return entry_list
+
+        def data_attribute_function(labels, context, desc, inst, name):
+            attribute_list = []
+            new_dict = {}
+            new_dict["semantics"] = self.context.pmSemStr(desc.contents.sem)
+            new_dict["type"] = get_type_string(desc)
+            if desc.indom != PM_INDOM_NULL:
+                new_dict["instname"] = name
+                new_dict["instid"] = inst
+            for key in labels:
+                new_dict.update(labels[key])
+            tmp_list = attribute_converter(new_dict)
+            for item in tmp_list:
+                attribute_list.append(item)
+            return attribute_list
+
+        def data_points_function(metric, results, labels, context, desc):
+            numdatapoints_list = []
+            for inst, name, value in results[metric]:
+                datapoint_dict = {}
+                tmp_type = get_type_string(desc)
+                if '32' in tmp_type or '64' in tmp_type:
+                    datapoint_dict["asInt"] = value
+                else:
+                    datapoint_dict["asDouble"] = value
+                if inst != PM_INDOM_NULL:
+                    datapoint_dict["instance"] = inst
+                datapoint_dict["startTimeUnixNano"] = ts
+                datapoint_dict["attributes"] = data_attribute_function(labels, context, desc, inst, name)
+                numdatapoints_list.append(datapoint_dict)
+            return numdatapoints_list
+
+        def attribute_function():
+            attribute_list = []
+            body = {}
+            body["os.type"] = platform.system().lower()
+            body["telemetry.sdk.language"] = "python"
+            body["telemetry.sdk.name"] = "opentelemetry"
+            body["telemetry.sdk.version"] = "1.24.0"
```

Comment:
Hmm?  We're not using the SDK though - so why put a version here?

We also should not hard-code the os.type like this, as a the tool may be sending data from an archive which was not created on Linux.

As mentioned elsewhere, I think we should treat the PCP context labels as the OTEL resource attributes - this seems like a close match, and means we can reduce the size of he generated JSON significantly (otherwise, these labels are re-produced for every instance of every metric).